### PR TITLE
OS-Agent support

### PR DESCRIPTION
**Pull request by pvizeli**
## Proposed change

Add basic OS Agent support for follow-up PRs. Each of those functions is so different that I split the implementation into multiple PRs and only add the dbus interface.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/991
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/host/__init__.py [hunk 1]
```diff
@@ -85,8 +85,11 @@ def supported_features(self) -> List[HostFeature]:
         if self.sys_dbus.timedate.is_connected:
             features.append(HostFeature.TIMEDATE)
 
+        if self.sys_dbus.agent.is_connected:
+            features.append(HostFeature.AGENT)
+
         if self.sys_hassos.available:
-            features.append(HostFeature.HASSOS)
+            features.append(HostFeature.HAOS)
```
Comment (ludeeus) [inline supervisor/host/__init__.py:92]: This will be breaking for the frontend, maybe more

Comment (pvizeli) [reply supervisor/host/__init__.py:92]: Right, but should be only frontend. I can make the naming of the string still hassos until we fixed the UI

Comment (ludeeus) [reply supervisor/host/__init__.py:92]: If it's only the UI, it's fine, we can just fix that before this is released.

Comment (ludeeus) [reply supervisor/host/__init__.py:92]: Let's kick legacy out, will finish review of this PR later today, and can prepare a frontend PR after that

File: supervisor/dbus/core.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+from .logind import Logind
+from .network import NetworkManager
+from .rauc import Rauc
+from .systemd import Systemd
+from .timedate import TimeDate
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class DBusManager(CoreSysAttributes):
```
Comment (ludeeus) [inline supervisor/dbus/core.py:19]: This is also called manager, is core the best name?
ref: https://github.com/home-assistant/supervisor/pull/2940#discussion_r659724749

File: supervisor/dbus/agent/__init__.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    @property
+    @dbus_property
+    def diagnostics(self) -> bool:
+        """Return if diagnostics is enabled on OS-Agent."""
+        return self.properties[DBUS_ATTR_DIAGNOSTICS]
+
+    @diagnostics.setter
+    def diagnostics(self, value: bool) -> None:
+        """Enable oder disable OS-Agent diagnostics."""
```
Comment (ludeeus) [inline supervisor/dbus/agent/__init__.py:74]:
```suggestion
        """Enable or disable OS-Agent diagnostics."""
```

File: supervisor/dbus/agent/__init__.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        )
+
+    async def connect(self) -> None:
+        """Connect to system's D-Bus."""
+        try:
+            self.dbus = await DBus.connect(DBUS_NAME_HAOS, DBUS_OBJECT_HAOS)
+            await self.cgroup.connect()
+            await self.apparmor.connect()
+            await self.system.connect()
+            await self.datadisk.connect()
```
Comment (ludeeus) [inline supervisor/dbus/agent/__init__.py:86]: Should we have `self.diagnostics = self.sys_config.diagnostics`  as part of the connection here?
Or do you want to handle that elsewhere?

File: supervisor/dbus/agent/apparmor.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        return (
+            await self.dbus.AppArmor.LoadProfile(profile.as_posix(), cache.as_posix())
+        )[0]
+
+    @dbus_connected
+    async def unload_profile(self, profile: Path, cache: Path) -> bool:
+        """Remove AppArmor profile."""
+        return (
+            await self.dbus.AppArmor.UnloadProfile(profile.as_posix(), cache.as_posix())
+        )[0]
```
Comment (ludeeus) [inline supervisor/dbus/agent/apparmor.py:52]: Why are these returning?
Nothing is checking their value

File: supervisor/dbus/agent/cgroup.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    """CGroup object for OS Agent."""
+
+    async def connect(self) -> None:
+        """Get connection information."""
+        self.dbus = await DBus.connect(DBUS_NAME_HAOS, DBUS_OBJECT_HAOS_CGROUP)
+
+    @dbus_connected
+    async def add_devices_allowed(self, container_id: str, permission: str) -> bool:
+        """Update cgroup devices and add new devices."""
+        return (await self.dbus.CGroup.AddDevicesAllowed(container_id, permission))[0]
```
Comment (ludeeus) [inline supervisor/dbus/agent/cgroup.py:19]: Why is this returning?

File: supervisor/dbus/agent/datadisk.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    @dbus_connected
+    async def update(self):
+        """Update Properties."""
+        self.properties = await self.dbus.get_properties(DBUS_NAME_HAOS_DATADISK)
+
+    @dbus_connected
+    async def change_device(self, device: Path) -> bool:
+        """Load/Update AppArmor profile."""
+        return (await self.dbus.DataDisk.ChangeDevice(device.as_posix()))[0]
```
Comment (ludeeus) [inline supervisor/dbus/agent/datadisk.py:41]:
```suggestion
        """Change device."""
        return (await self.dbus.DataDisk.ChangeDevice(device.as_posix()))[0]
```
Why is this returning?

File: supervisor/dbus/agent/system.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    """System object for OS Agent."""
+
+    async def connect(self) -> None:
+        """Get connection information."""
+        self.dbus = await DBus.connect(DBUS_NAME_HAOS, DBUS_OBJECT_HAOS_SYSTEM)
+
+    @dbus_connected
+    async def schedule_wipe_device(self) -> bool:
+        """Schedule a factory reset on next system boot."""
+        return (await self.dbus.System.ScheduleWipeDevice())[0]
```
Comment (ludeeus) [inline supervisor/dbus/agent/system.py:19]: Why is this returning?

File: supervisor/host/__init__.py (hunk 1 above)
Comment (ludeeus) [reply supervisor/host/__init__.py:92]: Frontend PR: <https://github.com/home-assistant/frontend/pull/9477>
Docs need update <https://developers.home-assistant.io/docs/api/supervisor/endpoints#host>

File: supervisor/dbus/agent/__init__.py (hunk 4 above)
Comment (pvizeli) [reply supervisor/dbus/agent/__init__.py:86]: right, will be later implemented on config + core setup process. For now, it just builds the bus interface

File: supervisor/host/__init__.py (hunk 1 above)
Comment (pvizeli) [reply supervisor/host/__init__.py:92]: done

File: scripts/supervisor.sh [hunk 9]
```diff
@@ -88,6 +88,21 @@ function init_udev() {
...
 }
 
+function init_os-agent() {
+    if pgrep os-agent; then
+        echo "os-agent is running"
+        return 0
+    fi
+
+    if [ ! -f /usr/sbin/os-agent ]; then
+        curl -Lo /usr/sbin/os-agent https://github.com/home-assistant/os-agent/releases/download/1.1.0/os-agent-debian-amd64.bin
```
Comment (ludeeus) [inline scripts/supervisor.sh:98]:
```suggestion
        curl -Lo /usr/sbin/os-agent https://github.com/home-assistant/os-agent/releases/latest/download/os-agent-debian-amd64.bin
```
Let's always use the latest tag so we don't have to update this file on each release

**Review by ludeeus — APPROVED** [review]

**Pull request merged**